src/timeline/index.ts

```diff
@@ -0,0 +1,114 @@
+import createOpen from './before/create';
+import * as log from '../log';
+
+
+export type condition = (events:TimelineEvent[], event?:TimelineEvent) => boolean;
+
+export const getTime = 'now' in performance ? () => {
+    return performance.now()
+} : () => {
+    return (new Date()).getTime()
+};
+
+const beforeTest:condition[] = [createOpen];
+const afterTest = [];
+
+class Timeline {
+    private events:TimelineEvent[];
+    private isRecording:boolean;
+    /** @constructor */
+    constructor() {
+        this.events = [];
+        this.isRecording = false;
+        // @ifdef DEBUG
+        try {
+            if (window !== window.parent && parent['__t'].isRecording) {
+                log.print('parent is recording');
+                this.isRecording = true;
+            }
+        } catch(e) { }
+        // @endif
+        this.registerEvent(new TimelineEvent(TLEventType.CREATE, undefined, undefined));
+    }
+    registerEvent(event:TimelineEvent):void {
+        let i = afterTest.length;
+        while (i--) { afterTest[i](this.events, event); }
+        this.events.push(event);
+        if (!this.isRecording) {
+            setTimeout(function() {
+                this.events.splice(this.events.indexOf(event), 1);
+            }.bind(this), 5000);
+        } else {
+            let name:string = event.name ? event.name.toString() : '';
```

Comment:
Basically, debug code work is different from the release one. This will cause problems in the future, so let's get rid of it right away.